packages/flutter_tools/lib/src/web/chrome.dart

```diff
@@ -117,7 +117,7 @@ class ChromeLauncher {
       }
     }
 
-    final int port = await globals.os.findFreePort();
+    final int port = debugPort ?? await globals.os.findFreePort();
```

Comment:
Should be some existing tests for chrome, you'll want to verify that we use the right port in the launch command. I recomend something with FakeProcessManager.list(<FakeCommand>[...expected launch command])

Reply:
Done.